apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp

```diff
@@ -711,6 +711,17 @@
                     return false;
                 }
 
+                var firstname = $("#firstname").val();
+                var lastname = $("#lastname").val();
+
+                if (!firstname.trim() || !lastname.trim()) {
+                    error_msg.text("<%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle,
+                        "Invalid.firstname.or.lastname")%>");
+                    error_msg.show();
+                    $("html, body").animate({scrollTop: error_msg.offset().top}, 'slow');
+                    return false;
+                }
```

Comment:
Can we have separate error messages for first name and last name?

Reply:
And update the messages to something like: "First name cannot be empty, please enter a valid First name"

Reply:
Resolving with https://github.com/wso2/identity-apps/pull/2504/commits/0bc394fbffde8cdbd88de647a3e7cf0904c50bc7